src/libs/actions/Policy.js

```diff
@@ -336,21 +399,31 @@ function invite(logins, welcomeNote, policyID) {
             // Save the personalDetails for the invited user in Onyx
             if (data.jsonCode === 200) {
                 Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, PersonalDetails.formatPersonalDetails(data.personalDetails));
-                Navigation.goBack();
+                const nextPolicy = _.clone(allPolicies[key]);
+                const nextEmployeeList = _.intersection(newEmployeeList, nextPolicy.pendingInvitations);
+                nextPolicy.employeeList = _.uniq([...nextPolicy.employeeList, ...nextEmployeeList]);
+                nextPolicy.pendingInvitations = _.difference(nextPolicy.pendingInvitations, nextEmployeeList);
+                Onyx.set(key, nextPolicy);
+
```

Comment:
Let's revert this if block to `data.jsonCode !== 200` and then just throw the received error and handle everything in catch callback. 
We can remove `revertPendingInvitationsList` inner function definition and move everything to catch.